### PR TITLE
[refactor] Remove with_entries() and keep_raw from Matrix

### LINKED ISSUE #2718: [refactor] Clean up keep_raw
Few places use `keep_raw` from the python Matrix class constructor. Currently, the parameter list is too long. It should be great to remove `keep_raw` option and find an alternative way to do so.

https://github.com/taichi-dev/taichi/blob/4d55957214e5c33aac1d68c7b2dbba5b0567d777/python/taichi/lang/ndrange.py#L45-L47

### PR DESCRIPTION
**Pull request by strongoier**
Related issue = #2880

This is a follow-up PR of #3536. Meanwhile, it fixes #2718 by the way.

### CODE REVIEW

File: python/taichi/lang/matrix.py [hunk 1]
```diff
@@ -1146,6 +1127,25 @@ def Vector(n, dt=None, **kwargs):
 Vector.normalized = Matrix.normalized
 
 
+class IntermediateMatrix(Matrix):
```
Comment (ailzhang) [inline python/taichi/lang/matrix.py:1130]: nit: `_IntermediateMatrix` :P ?

**Review by ailzhang — APPROVED** [review]
LGTM! Thanks!

File: python/taichi/lang/matrix.py (hunk 1 above)
Comment (strongoier) [reply python/taichi/lang/matrix.py]: Sounds good. Done.

**Pull request merged**